### ISSUE TITLE
[FEATURE] Add SQLITE3 ODBC to all the kicad* containers

### ISSUE THREAD

**Issue by hpopols** [INTI-CMNB/kicad_debian#3]
Hi everybody,

It would be nice to see SQLITE3 ODBC included in all the containers in order to use the 'new' database library available in kicad.

Following the following [guide](https://www.dbi-services.com/blog/installing-the-odbc-driver-manager-with-sqlite-on-linux/), I am currently installing it after the kicad7_auto container creation with:
```
apt update
apt install libodbc1 -y
apt install unixodbc -y
odbcinst -j
export LD_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu/odbc:$LD_LIBRARY_PATH

apt install sqlite3 -y
apt install libsqliteodbc -y
```
Remark: I am not sure if the `odbcinst -j` is needed but the LD_LIBRARY_PATH has to be done before the libsqliteodbc install for it to register properly.

I can imagine other ODBC driver could also be usefull (MariaDB, Postgresql,...)...

Thank you for this amazing work.

Xavier

**Comment by set-soft** [comment]
Hi @hpopols !

This looks interesting, but:
1. If added it will be added to *kicad7_auto_full*, not *kicad7_auto* because this is an uncommon tool. Less than 1% users will really use it.
2. The guide looks really strange to me. 

Why it looks strange?
- I don't see why a package will need to manually run **odbcinst -j**, this should be run in the *postinst* script.
- Why do you need to set an **LD_LIBRARY_PATH**? This is needed only if some of the components isn't really compiled for the base system and you need to force something. Otherwise the code should look for _plugins_ there.

Looking at Debian descriptions:
- **unixodbc**: This package contains isql, a command-line tool that allows SQL commands to be entered interactively.
Is this really needed? Also note that this will pull *libodbc1* as a dependency.
- **sqlite3**: Is this really needed? *libsqliteodbc* pulls the *libsqlite3-0* package.

And looking at **libsqliteodbc** I see:

``` 
odbcinst -i -d -f /usr/share/sqliteodbc/unixodbc.ini
```

So the tutorial you used doesn't look good to me. May be it works, but does a lot of things that doesn't seem to be needed. Try only doing this:

```
apt update
apt install libodbc1 libsqliteodbc -y
```

And let me know if provides the same functionality

**Comment by hpopols** [comment]
Hi @set-soft !

> 1. If added it will be added to _kicad7_auto_full_, not _kicad7_auto_ because this is an uncommon tool. Less than 1% users will really use it.

I agree at least for now because this functionality in kicad is pretty new and is not very polished yet (only basic functionalities are now implemented, there is performance issues and the documentation is very light). However, I have the feeling people using CI/CD and kicad_auto are likely to use database libs in the future (but perhaps not SQLite3 ones). Futhermore, database libs are widely used with most common "professional" CAD software in the electronic industry and good/bad habits are difficult to change.

My choice for SQLite was mainly driven by 2 things:

1. Can be versioned as binary files and included in the kicad project as a git submodule
3. Is able to work offline (opposed to classic databases)

Concerning the ODBC installation:
> So the tutorial you used doesn't look good to me. May be it works, but does a lot of things that doesn't seem to be needed. Try only doing this:
> 
> ```
> apt update
> apt install libodbc1 libsqliteodbc -y
> ```
> 
> And let me know if provides the same functionality

After digging a little into odbc and the debian packages, I realized you were right!

I did test your proposition in a _kicad7_auto:1.6.3_ container and it does not work but it was pretty close. Indeed, the SQLite3 driver (from _libsqliteodbc_) has to be installed (registrered) to ODBC with the _odbcinst_ tool to work. However this tool is not installed by default...

So the working solution is a simple:
```
apt update
apt install odbcinst libodbc1 libsqliteodbc -y
```

From there the ODBC driver for SQLite is now known as 'SQLite3'.

For reference the "source" parameter in the library database config file _*.kicad_dbl_ should look like:
```
...
    "source": {
        "type": "odbc",
        "dsn": "",
        "username": "",
        "password": "",
        "timeout_seconds": 2,
        "connection_string": "Driver={SQLite3};Database=database.sqlite"
    },
...
```
